lib/handover.js

```diff
@@ -68,58 +72,52 @@ class Plugin {
         return res;
 
       let hasEnsReferral = false;
-      // Check NS records for referals to TLDs `.eth` and `._eth`
+      // Check NS records for referrals to TLDs `.eth` and `._eth`
       for (const rr of res.authority) {
         if (rr.type !== wire.types.NS)
           continue;
 
+        const ending = util.label(rr.data.ns, util.split(rr.data.ns), -1);
+
         // Look up the ENS resolver specified in the NS record
         // and query it for the user's original request
-        if (rr.data.ns.slice(-5) === '.eth.') {
+        if (ending === '_eth' || ending === 'eth') {
```

Comment:
I like how you refactored these into a group so `resolveDnsFromEns()` and `resolveDnsFromAbstractEns()` are next to each other down below. :+1: